README.md

```diff
@@ -1,57 +1,179 @@
+* Quickstart: Azure CLI based step-by-step scripts with a simplified setup (single Virtual Network & Resource Group) and sample client app
 
 ## Getting Started
 
 ### Prerequisites
 
-(ideally very short, if any)
+* Azure subscription
+* Azure CLI
+* Visual Studio Code
+* OpenSSL CLI
+* Bash terminal, on Windows you can use WSL
+* .NET SDK 8 (if you wish to run the client and server samples for file upload)
+* DNS provider if you wish to test HTTPS with your custom domain, and SSL Certificate for your custom domain
 
-- OS
-- Library version
-- ...
+> [!TIP]
+> Without a custom DNS and an SSL certificate available, you can still test the entire setup but the IoT client device portion of this quickstart will not be completed.
 
-### Installation
+### Quickstart
 
-(ideally very short)
+This quickstart provides instructions on how to set up Azure IoT Hub's file upload functionality through an Azure Storage account that only allows private connections, through step-by-step scripts. The scripts deploy and configure a simplified set of Azure resources to showcase Azure Storage with private endpoint, Application Gateway and Azure IoT Hub routing to Azure Storage. The Storage account disables any public Internet access and only accessible within the Virtual Network. In this quickstart there is no Azure Firewall for traffic inspection.
+Finally, two sample .NET apps interact with the resources deployed to showcase the end to end data flows.
 
-- npm install [package name]
-- mvn install
-- ...
+> [!WARNING]
+> This sample currently generates self-signed certificate and key files on disk. This should be improved to leverage Azure Key Vault for securely storing and retrieving the password, certificate and key. Application Gateway supports retrieving an SSL certificate from a Key Vault account.
 
-### Quickstart
-(Add steps to get up and running quickly)
+#### Setup
 
-1. git clone [repository clone url]
-2. cd [repository name]
-3. ...
+* Open a Bash terminal
+* `git clone` this repo
+* cd [repository name]
+* Open the project in Visual Studio Code `code .` and open a Bash terminal.
+* Log into your Azure account `az login [--tenant xxxx-xxx]` and set the default subscription `az account set -s <subscription>`
+* Prepare required environment variables to run the deployment scripts. We create a file in the folder `./temp` which is excluded from Git.
 
+  * Run the below to create the file `./temp/envvars.sh`, for `LOCATION` you can choose any Azure region supporting IoT Hub and Application Gateway.
 
-## Demo
+    ```bash
+    if [ ! -d "./temp" ]; then
+        mkdir ./temp
+    fi
+    >./temp/envvars.sh cat <<EOF
+    # change the below set to match your environment based on Readme
+    export TENANT_ID="xxx-xxxx-xxx-xxxx-xxxx"
+    export CERT_PASSWORD="xxx"
```

Comment:
Please don't demonstrate, even in development, even with a fake cert, a cert password in a file.  Have the script prompt the user as part of the deployment steps.  This is an SFI (teach security) requirement.

Reply:
This is no longer part of the repo, it's from an early version